### ISSUE TITLE
License deploy to asset issue

### ISSUE THREAD

**Issue by chansinshan** [snipe/snipe-it#4589]
#### Expected Behavior (or desired behavior if a feature request)

An asset was deployed to a location.  And the license is checked out to this asset.  The license only show the asset ID and the user field is blank.

-----

#### Actual Behavior

The asset show the user name in the user field with the asset ID.

-----

#### Please confirm you have done the following before posting your bug report:

- [  ] I have enabled debug mode 
- [ x ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Provide answers to these questions:

- Is this a fresh install or an upgrade?
Upgrade
- Version of Snipe-IT you're running
v4.1.8-pre - build 3034 (master)
- Version of PHP you're running
	PHP Version 7.0.22
- Version of MySQL/MariaDB you're running
	mariadb-server-10.1
- What OS and web server you're running Snipe-IT on
	Ubuntu 16.04.3 LTS / PHP Version 7.0.22
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc)
	Install.sh
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
	no
- Confirm whether the error is reproduceable on the demo: https://snipeitapp.com/demo.
	Cannot reproduce this error, because the asset cannot check out to location
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

https://snipe-it.readme.io/docs/getting-help

**Comment by Diniver** [comment]
I am able to reproduce it as well.

**Comment by chansinshan** [comment]
Hi @Diniver , do you mean you can reproduce the same error result?

**Comment by Diniver** [comment]
yes. Even in activity history when I checked the license in, it shows that the license was checked in from a user not an asset. 
![image](https://user-images.githubusercontent.com/31882138/33707488-68fb8d88-db2f-11e7-94fa-b7bb9377eb73.png)

**Comment by snipe** [comment]
I don't understand this issue - can someone clarify?

**Comment by Diniver** [comment]
No problem. It took me some time to get it as well. So: If asset named _PC_ is checked out to location _Room1_ and after that you give license _Office_ to this asset (PC) then your license is assigned to a user as well. In my case Tihomir. The idea is that user field to stay blank when you check out licenses to assets

![image](https://user-images.githubusercontent.com/31882138/33710651-2b4994c6-db39-11e7-8758-125270aa7495.png)

**Comment by chansinshan** [comment]
@Diniver Thank you for your explain.

**Comment by chansinshan** [comment]
@snipe I tried checkin the license and the asset.  Then checkout the license to this asset again.  And the asset still remain "Ready for checkout" status.  The license will return to normal only display the asset ID without user ID.  

That means if the asset is checkout for location or asset.  The license's error will come out.

Hope you understand my bad English.

**Comment by snipe** [comment]
> The idea is that user field to stay blank when you check out licenses to assets

If the asset is checked out to that user though, how is that not a good thing to display?

**Comment by chansinshan** [comment]
I tested on the your demo site.  If I assigned a license to an asset.  The license user field will show as the asset deployed location.

But my self hosted system is not show the asset deployed location.  Is mistake to show the user name.  And that user is not related to the asset.

**Comment by snipe** [comment]
So you cannot reproduce it on the demo?

**Comment by chansinshan** [comment]
Yes, I just checked out the asset "460674071" to location "Amaraview".
![default](https://user-images.githubusercontent.com/26160283/33812193-3e7f3878-de56-11e7-92c6-db79ac7a420b.png)

Then I deployed the office license to this asset and it show the user as "Braxton Carroll".  But this asset is not related to this user.
![default](https://user-images.githubusercontent.com/26160283/33812231-847bf8a2-de56-11e7-9154-70f62eeb4e50.png)
